src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -113,28 +96,31 @@ public void storeMessage(Message message) {
     // All the messages between the two users
     Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
 
+    // Sort in the ascending order so that most recent messages appear last
     Query query =
         new Query("Message")
             .setFilter(directMessages)
             .addSort("timestamp", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
+
+      String idString = null;
       try {
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String text = (String) entity.getProperty("text");
-        long timestamp = (long) entity.getProperty("timestamp");
-        String sender = (String) entity.getProperty("user");
-        String receiver = (String) entity.getProperty("receiver");
-        Message message = new Message(id, sender, text, timestamp, receiver);
-
-        messages.add(message);
-      } catch (Exception e) {
+        idString = entity.getKey().getName();
+      } catch (NullPointerException e) {
         System.err.println("Error reading message.");
         System.err.println(entity.toString());
         e.printStackTrace();
       }
+
+      UUID id = UUID.fromString(idString);
+      String text = (String) getStringProperty(entity, "text").orElse("");
+      long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
Can you create a similar `getLongProperty` method?

Reply:
Is this necessary? Since the queries are ordered by timestamp earlier in the method, at this point should we assume that a timestamp does in fact exist? If it didn't, would the earlier query return an error?

Reply:
This is also why I didn't use the `getStringProperty` method for retrieving the `user` and `recipient` because the earlier query would ensure that all the messages had the user and recipient I'm looking for.

Reply:
That's a fair point. No, it isn't necessary.